### ISSUE TITLE
Repair on "::class can be used" does not detect collision with current class

### ISSUE THREAD

**Issue by MorrisJobke** [kalessil/phpinspectionsea#770]
### Steps

Having two classes:

`src/Abc/MyClass.php`:

```php
<?php

namespace OC\Abc;

class MyClass {
	public function abc() {
		echo '\OC\Abcd\MyClass';
	}
}
```

`src/Abcd/MyClass.php`:

```php
<?php

namespace OC\Abcd;

class MyClass {
}
```

The inspection `::class can be used` is shown:

<img width="1081" alt="bildschirmfoto 2018-01-26 um 12 13 59" src="https://user-images.githubusercontent.com/245432/35437376-701e27f8-0292-11e8-827a-438974d58ae7.png">

### Problem

Clicking the repair button now results in:

```php
<?php

namespace OC\Abc;

use OC\Abcd\MyClass;

class MyClass {
	public function abc() {
		echo MyClass::class;
	}
}
```

This is obviously wrong, because `MyClass` is already defined. It seems that it does not detect collisions with the class it is currently repairing.

### Possible solution no. 1

The code needs to be either the full name:

```php
<?php

namespace OC\Abc;

class MyClass {
	public function abc() {
		echo \OC\Abcd\MyClass::class;
	}
}
```

### Possible solution no. 2

or aliased:

```php
<?php

namespace OC\Abc;

use OC\Abcd\MyClass as AbcdMyClass;

class MyClass {
	public function abc() {
		echo AbcdMyClass::class;
	}
}
```

**Comment by kalessil** [comment]
Thank you for reporting @MorrisJobke, we'll follow the `Possible solution no. 1` strategy here.

**Comment by kalessil** [comment]
Fixed
